simulation/traffic_simulator/src/traffic_lights/traffic_lights_base.cpp

```diff
@@ -123,6 +123,10 @@ auto TrafficLightsBase::addTrafficLight(const lanelet::Id traffic_light_id) -> v
 
 auto TrafficLightsBase::getTrafficLight(const lanelet::Id traffic_light_id) -> TrafficLight &
 {
+  if (isTrafficLightAdded(traffic_light_id)) {
+    return traffic_lights_map_.at(traffic_light_id);
+  }
+
   addTrafficLight(traffic_light_id);
   return traffic_lights_map_.at(traffic_light_id);
```

Comment:
Thank you for contributing!
This may be better.
```suggestion
  if (not isTrafficLightAdded(traffic_light_id)) {
    addTrafficLight(traffic_light_id);
  }
  return traffic_lights_map_.at(traffic_light_id);
```

Reply:
Make sense. I fixed it in https://github.com/tier4/scenario_simulator_v2/pull/1499/commits/1e5d510785f6117235bd9231c4b3ba2fc62f1328.